etl/steps/data/meadow/irena/2023-12-12/renewable_power_generation_costs.py

```diff
@@ -63,36 +59,34 @@ def prepare_solar_pv_module_prices(excel_object: pr.ExcelFile) -> Table:
     pv_prices["year"] = pd.to_datetime(pv_prices["month"], format="%b %y").dt.year
 
     # For each year get the average cost over all months.
-    # NOTE: The following operation does not properly propagate metadata. Do it manually.
-    _pv_prices = pv_prices.copy()
     pv_prices = (
         pv_prices.groupby(["technology", "year"])
         .agg({"cost": "mean", "year": "count"})
         .rename(columns={"year": "n_months"})
         .reset_index()
-    ).copy_metadata(_pv_prices)
+    )
 
     # Remove unnecessary column and add column for region.
-    pv_prices = pv_prices.drop(columns="technology").assign(**{"country": "World"})
+    pv_prices = pv_prices.drop(columns="technology", errors="raise").assign(**{"country": "World"})
 
     # Ignore years for which we don't have 12 months.
-    pv_prices = pv_prices[pv_prices["n_months"] == 12].drop(columns=["n_months"]).reset_index(drop=True)
+    pv_prices = pv_prices[pv_prices["n_months"] == 12].drop(columns=["n_months"], errors="raise").reset_index(drop=True)
```

Comment:
I'd check that this works as expected (e.g. assertion error that only the most recent year is missing bc there is no data for 12 months; if there are other years for when that's the case it's good to add it to your metadata descriptions of the variables later on)